Makefile-Tasking-Tricore/custom_compiler_yaml_tasking.py

```diff
@@ -0,0 +1,70 @@
+output_filename = "custom-compiler-tasking-generated.yaml"
+tasking_install_folder = "C:\\Program Files\\TASKING\\TriCore v6.3r1"
+tasking_platform_suffix = ["51", "arm", "mcs", "pcp", "tc"]  # TriCore, ARM
+
+c_additional_compile_switches = "--iso=99"  # example, may be empty
+
+cpp_additional_compile_switches = "--c++=11"  # example, may be empty
+
+import os, subprocess, tempfile, re
+
+
+def extract_defines(compiler_bin, src_name):
+    result = subprocess.run([compiler_bin, "-El", src_name], stderr=subprocess.STDOUT, stdout=subprocess.PIPE)
+    if result.returncode != 0:
+        return None
+    all_defines = result.stdout.splitlines()
```

Comment:
I'd decode here instead of `b'...'`: `result.stdout.decode('utf-8').splitlines()`

Reply:
Also there is subprocess.check_output, to execute safely and get output
```
output = subprocess.check_output(
      [compiler_bin, "-El", src_name]
).decode().strip().splitlines()
```

In case of files i would prefer writes instead of prints, feels more, hm, natural i guess, you don't pass anything but single argument strings anyways.

Reply:
@dancing-leaves I don't want any exceptions to be raised here. In some cases (ex. 8051) the compiler does not support c++ at all, and that is the valuable result as well, which is checked later.

Reply:
@fastasturtle What about https://www.trojansource.codes/

Reply:
Changed the code a little